haskell/private/cc_wrapper.py.tpl

```diff
@@ -134,12 +142,22 @@ class Args:
         self.args = list(self._handle_args(args))
 
         if self.linking:
+            # GHC with Template Haskell or tools like hsc2hs build temporary
+            # Haskell binaries which we make sure are linked in accordance with
+            # the wider build.
             if is_temporary_output(self.output):
-                # GHC with Template Haskell or tools like hsc2hs builds
-                # temporary Haskell binaries linked against libraries, but does
-                # not speficy the required runpaths on the command-line in the
-                # context of Bazel.
+                # Temporary binary build commands do not specify required
+                # runpaths on the command line in the context of Bazel; fix
+                # this explicitly.
                 self.rpaths.extend(self.library_paths)
+
+            # When in a static configuration, we need to ensure that the objects
+            # that hsc2hs builds in particular are statically linked. Since the
+            # static configuration should ensure that all the objects involved
+            # are themselves statically built, this is just a case of passing
+            # `-static` to the real C compiler to produce the final object.
+            if self.output.endswith("hsc_make") or self.output.endswith("hsc_make.o"):
+                self.args.extend(["-static"])
```

Comment:
As mentioned in https://github.com/tweag/rules_haskell/issues/379#issuecomment-657456884 I will see if this can be hoisted to a higher level.